pkg/kubelet/network/dns/dns.go

```diff
@@ -389,7 +389,9 @@ func (c *Configurer) GetPodDNS(pod *v1.Pod) (*runtimeapi.DNSConfig, error) {
 				dnsConfig.Servers = append(dnsConfig.Servers, ip.String())
 			}
 			dnsConfig.Searches = c.generateSearchesForDNSClusterFirst(dnsConfig.Searches, pod)
-			dnsConfig.Options = defaultDNSOptions
```

Comment:
If I'm understanding this change correctly, this would mean we inhereit DNS options (i.e. ndots) from the host, which I think would be a significant breaking change in the current default behavior we have today that assumes ndots:5, which is important for cluster DNS.

Reply:
We probably want some API-driven way to configure this default though, I'm not sure inheriting from host is safe at this point

Reply:
> which I think would be a significant breaking change in the current default behavior

Yeah I pointed this out in the description. I also called this out as a bug fix because it seems strange and unfair to me as a cluster administrator for `kubelet` to offer a `--resolvconf` but then to explicitly override the fields that I want pods to be configured with.  

> We probably want some API-driven way to configure this default though

What about a `kubelet` flag like `--resolveConfOptions` where the string passed there would override the `defaultDNSOptions`? I'm okay with removing the behavior change here, making this PR a simple test improvement (there are valid test cases added here that weren't covered before), then opening a PR to make it possible to override `defaultDNSOptions` in whatever way you prefer.

>  I'm not sure inheriting from host is safe at this point

Would it ever be considered safe?

Reply:
> the current default behavior we have today that assumes ndots:5

One thing I don't understand is why 5 was chosen here to begin with. Are cluster-local hostnames expected to reach that high a number of dots? For what it's worth, I spend more time developing and administrating cluster provisioning backends than actually using Kubernetes myself so it's not clear what would lead cluster local hostnames to have up to 5 dots.

Reply:
I was thinking about this last night and realized that the `ndots:5` option only applies for `DNSClusterFirst` and `DNSClusterFirstWithHostNet` (when host network is false). It's in the name, "cluster first" -- that explains the absurdly large `ndots` value that ensures even obviously absolute/non-cluster hostnames should be DNS searched before being treated as absolute during lookup.

Reply:
Yes, exactly -- this is specific behavior only when going through cluster DNS (kube-dns/coredns)